logical/src/main/java/org/apache/drill/common/expression/SchemaPath.java

```diff
@@ -334,36 +335,13 @@ public int hashCode() {
 
   @Override
   public boolean equals(Object obj) {
-    if (this == obj) {
-      return true;
-    }
-    if (obj == null) {
-      return false;
-    }
-    if (!(obj instanceof SchemaPath)) {
-      return false;
-    }
-
-    SchemaPath other = (SchemaPath) obj;
-    if (rootSegment == null) {
-      return (other.rootSegment == null);
-    }
-    return rootSegment.equals(other.rootSegment);
+    return this == obj || obj instanceof SchemaPath
+        && Objects.equals(rootSegment, ((SchemaPath) obj).rootSegment);
   }
 
   public boolean contains(Object obj) {
```

Comment:
`contains()` is not a Java-defined method. Must the signature be `Object`? Or, can it be `SchemaPath` since that is the only kind of thing a `SchemaPath` could ever contain? Would save some casting.